src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public static final int MAGIC_NUMBER = 47;
+    private final Random random = new Random(MAGIC_NUMBER);
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Figure[] figures = new Figure[]{new Circle(Color.valueOf(
+        colorSupplier.getRandomColor()), random.nextDouble()),
+        new IsoscelesTrapezoid(Color.valueOf(colorSupplier.getRandomColor()),
+            random.nextDouble(), random.nextDouble(), random.nextDouble()),
+        new Rectangle(Color.valueOf(colorSupplier.getRandomColor()),
+            random.nextDouble(), random.nextDouble()),
+        new Square(Color.valueOf(colorSupplier.getRandomColor()), random.nextDouble())};
```

Comment:
Instantiating figures with random parameters directly in the array initialization can lead to less readable code. It's better to create separate variables for the parameters and pass them to the constructors. Also, avoid creating figures in the array as class-level fields. Instead, create them inside the 'getRandomFigure()' method to ensure that each time the method is called, a new figure with new random parameters is returned.